### PR TITLE
Merge FLoC(FirebaseSegmentation) SDK code into master branch

### PR DESCRIPTION
**Pull request by diwu-arete**

### CODE REVIEW

File: firebase-segmentation/api.txt [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  public enum SetCustomInstallationIdException.Status {
+    enum_constant public static final com.google.firebase.segmentation.SetCustomInstallationIdException.Status BACKEND_ERROR;
+    enum_constant public static final com.google.firebase.segmentation.SetCustomInstallationIdException.Status CLIENT_ERROR;
+    enum_constant public static final com.google.firebase.segmentation.SetCustomInstallationIdException.Status DUPLICATED_CUSTOM_INSTALLATION_ID;
+    enum_constant public static final com.google.firebase.segmentation.SetCustomInstallationIdException.Status UNKOWN;
+  }
+
+}
+
+package com.google.firebase.segmentation.local {
```
Comment (vkryachko) [inline firebase-segmentation/api.txt]: Is this public api? if not pls add a `package-info.java` file with an `@hide` on it.

File: firebase-segmentation/api.txt [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    ctor public CustomInstallationIdCacheEntryValue();
+    method @NonNull public static com.google.firebase.segmentation.local.CustomInstallationIdCacheEntryValue create(@NonNull String, @NonNull String, @NonNull com.google.firebase.segmentation.local.CustomInstallationIdCache.CacheStatus);
+    method @NonNull public abstract com.google.firebase.segmentation.local.CustomInstallationIdCache.CacheStatus getCacheStatus();
+    method @NonNull public abstract String getCustomInstallationId();
+    method @NonNull public abstract String getFirebaseInstanceId();
+  }
+
+}
+
+package com.google.firebase.segmentation.remote {
```
Comment (vkryachko) [inline firebase-segmentation/api.txt]: same here and the rest of the packages

File: subprojects.cfg [hunk 3]
```diff
@@ -28,7 +28,7 @@ firebase-installations
 firebase-storage
 firebase-storage:ktx
 firebase-storage:test-app
-
+firebase-segmentation
```
Comment (vkryachko) [inline subprojects.cfg]: pls move it above `firebase-storage` to keep the lines sorted

File: firebase-segmentation/api.txt (hunk 1 above)
Comment (diwu-arete) [reply firebase-segmentation/api.txt]: Done

**Pull request merged**